images/build/cross/variants.yaml

```diff
@@ -35,30 +35,12 @@ variants:
     GO_MAJOR_VERSION: '1.23'
     OS_CODENAME: 'bullseye'
     REVISION: '0'
-  v1.31-go1.22-bullseye:
-    CONFIG: 'go1.22-bullseye'
-    TYPE: 'default'
-    IMAGE_VERSION: 'v1.31.0-go1.22.12-bullseye.0'
-    KUBERNETES_VERSION: 'v1.31.0'
-    GO_VERSION: '1.22.12'
-    GO_MAJOR_VERSION: '1.22'
-    OS_CODENAME: 'bullseye'
-    REVISION: '0'
-  v1.30-go1.22-bullseye:
-    CONFIG: 'go1.22-bullseye'
+  v1.30-go1.23-bullseye:
+    CONFIG: 'go1.23-bullseye'
     TYPE: 'default'
-    IMAGE_VERSION: 'v1.30.0-go1.22.12-bullseye.0'
+    IMAGE_VERSION: 'v1.30.0-go1.23.6-bullseye.0'
     KUBERNETES_VERSION: 'v1.30.0'
-    GO_VERSION: '1.22.12'
-    GO_MAJOR_VERSION: '1.22'
-    OS_CODENAME: 'bullseye'
-    REVISION: '0'
-  v1.29-go1.22-bullseye:
```

Comment:
Don't we need 1.29 go 1.23?

Reply:
It's EOL date isn't until the end of the month, I thought we should get it onto go 1.23 before we lock the branch

Reply:
Hrmm, really? It's pretty unfortunate to leave the last release build on an EOL go version, but I guess I don't feel that strongly

Reply:
what do y'all think @saschagrunert @xmudrii @Verolop @kubernetes/release-managers ?

Reply:
We could still do another 1.29 for that reason. I can remember we did that in the past.

Reply:
sgtm, i will add

Reply:
done